svg/src/circle.cpp

```diff
@@ -0,0 +1,21 @@
+#include <svg/element.hpp>
+
+#include "circle.hpp"
+
+namespace svg::geometry {
+
+    Circle::Circle(const geometry::Point& center, double radius):
+        center(center), radius(radius) {}
+
+    std::string Circle::to_string() const {
+        std::stringstream circle;
+
+        circle << start_element("circle");
+        circle << add_attribute("cx", center.get_x());
+        circle << add_attribute("cy", center.get_y());
+        circle << add_attribute("r", radius);
+        circle << end_element();
```

Comment:
If you introduce a XML element class, it's a prime example for a builder pattern. You could do something like:
```cpp
XMLNode node = NodeBuilder("tag")
  .with_attribute("key", "val")
  .build();
```
Just try to find maybe more appropriate type but you get the idea.. 
Notice that you don't have to specify the end tag which makes you code a lot more robust with no added complexity (the code is even more readable)